### DISCUSSION TITLE
Chunked media upload not working for large GIFs

### DISCUSSION THREAD

**Discussion by carlosgio** [tweepy/tweepy#1622] [Questions] [answered]
Hey, I need to use the chunked media upload so I'm using the latest development version:

`tweepy                   4.0.0a0`

But when I try to upload a 9 Mb GIF file (as per Twitter docs: _Image size <= 5 MB, animated GIF size <= 15 MB_) a 400 Bad Request is returned with no additional info:

```
>>>response_media_upload = api.media_upload(filename = "/home/carlos/repos/oneclout/gif_small.gif", chunked = True)
>>>response_media_upload = api.media_upload(filename = "/home/carlos/repos/oneclout/gif_big.gif", chunked = True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/carlos/.local/lib/python3.6/site-packages/tweepy/api.py", line 3403, in media_upload
    additional_owners=additional_owners, **kwargs
  File "/home/carlos/.local/lib/python3.6/site-packages/tweepy/api.py", line 3502, in chunked_upload
    additional_owners=additional_owners, **kwargs
  File "/home/carlos/.local/lib/python3.6/site-packages/tweepy/api.py", line 46, in wrapper
    return method(*args, **kwargs)
  File "/home/carlos/.local/lib/python3.6/site-packages/tweepy/api.py", line 3639, in chunked_upload_init
    upload_api=True, **kwargs
  File "/home/carlos/.local/lib/python3.6/site-packages/tweepy/api.py", line 255, in request
    raise BadRequest(resp)
tweepy.errors.BadRequest: 400 Bad Request
```

As you can see, GIFs smaller than 5 Mb are uploaded correctly. Any help would be much appreciated.

**Comment by Harmon758** [comment, marked as answer]
You need to pass `media_category`, e.g.:
```py
response_media_upload = api.media_upload(
    filename = "/home/carlos/repos/oneclout/gif_big.gif", chunked = True,
    media_category = "tweet_gif"
)
```

See https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/uploading-media/media-best-practices.

**Reply by carlosgio** [reply]
Thank you, that fixed the problem! 👏

For those who might get a similar problem in the future: even though [Tweepy documentation](https://docs.tweepy.org/en/latest/api.html#upload-media) specifies that _This field is required when using the media with the Ads API_, the ["Twitter documentation"](https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/uploading-media/media-best-practices) says that _In order to process larger GIFs, use the chunked upload endpoint with the media_category_.
